runtime/manifest.js

```diff
@@ -141,14 +141,22 @@ class Manifest {
   async newHandle(type, name, id, tags) {
     assert(!type.hasVariableReference, `handles can't have variable references`);
     let handle = await this.storageProviderFactory.construct(id, type, `in-memory://${this.id}`);
-    handle.name = name;
     assert(handle._version !== null);
+    handle.name = name;
+    return this._addHandle(handle, tags);
+  }
+
+  _addHandle(handle, tags) {
     this._handles.push(handle);
     this._handleTags.set(handle, tags ? tags : []);
     this._handleManifestUrls.set(handle.id, this.fileName);
     return handle;
   }
 
+  newHandleStub(type, name, id, storageKey, tags) {
```

Comment:
Isn't this variant meant to not populate handleManifestUrls?

Reply:
Uh .. yeah ..